src/extension_utils.c

```diff
@@ -95,7 +95,13 @@ extension_version(void)
 
 		if (!is_null)
 		{
-			sql_version = strdup(TextDatumGetCString(result));
```

Comment:
I think the correct fix is to replace `strdup` with `pstrdup` here and remove the `static` from `sql_version`.

Reply:
Likely, but if for some reason  HeapTupleIsValid() returns false with strdup the function value is the previously saved (stalled?) extension version, with pstrdup it kills the backend with elog(ERROR, ...)

 I'd expect sql_version being rather constant.

I have no idea if HeapTupleIsValid can return null but looking at the stack trying to get data from a multi processes cache being invalidated and doing it recursively in AcceptInvalidationMessages  call back, hum it's a race in waiting. 
As a matter of fact the recursion could be a bug in its own right, processing the same messages again and again or too early.

Anyway if you think it's ok I will replace it.

Reply:
_sigh_ yeah, the recursion looks like a bug in the making, good catch!

In the case that `!HeapTupleIsValid()` we don't have a valid version of the extension installed, so I think the error is correct; better than starting up the wrong version.